flink-core/src/main/java/org/apache/flink/core/fs/LimitedConnectionsFileSystem.java

```diff
@@ -313,14 +313,22 @@ public int getNumberOfOpenInputStreams() {
 	 * Get the rate limitation on Input (bytes/s).
 	 */
 	public long getRateLimitingInput(){
-		return (long) inputRateLimiter.getRate();
+		if (inputRateLimiter.isPresent()){
```

Comment:
Nitty nit (matter of taste): `return (long) inputRateLimiter.map(RateLimiter::getRate).getOrDefault(0)` if you prefer if/else I'm fine with that.

Btw, `Optional` saved the day. Previously there were some missed null pointer exceptions :)